File: src/main/java/com/wanted/preonboarding/reservation/application/dto/ReservationResponse.java
package com.wanted.preonboarding.reservation.application.dto;

import com.wanted.preonboarding.common.model.SeatInfo;
import com.wanted.preonboarding.performance.domain.entity.Performance;
import com.wanted.preonboarding.reservation.domain.dto.UserInfo;
import com.wanted.preonboarding.reservation.domain.entity.Reservation;
import lombok.AllArgsConstructor;
import lombok.Builder;

import java.util.UUID;

@Builder
@AllArgsConstructor
public class ReservationResponse {

    private final int round;

    private final String performanceName;

    private final UUID performanceId;

    private final SeatInfo seatInfo;

    private final UserInfo userInfo;

    public static ReservationResponse from(final Performance performance, final Reservation reservation) {
        return ReservationResponse.builder()
                .performanceId(performance.getId())
                .performanceName(performance.getName())
                .round(performance.getRound())
                .seatInfo(reservation.getSeatInfo())
                .userInfo(reservation.getUserInfo())
                .build();
    }
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/presentation/response/ReservationResponse.kt
package com.wanted.preonboarding.ticket.presentation.response

import com.wanted.preonboarding.ticket.domain.Performance
import com.wanted.preonboarding.ticket.domain.SeatInfo
import com.wanted.preonboarding.ticket.domain.UserInfo
import java.util.UUID

data class ReservationResponse(
    val performanceId: UUID,
    val name: String,
    val round: Int,
    val userInfo: UserInfo,
    val seatInfo: SeatInfo,
) {
    companion object {
        fun from(
            performance: Performance,
            userInfo: UserInfo,
            seatInfo: SeatInfo,
        ): ReservationResponse {
            return ReservationResponse(
                performanceId = performance.id.value,
                name = performance.name,
                round = performance.round,
                userInfo = userInfo,
                seatInfo = seatInfo,
            )
        }
    }
}
